### PR TITLE
[14.0][FIX] stock_request: Change state fields to display translated by language

### PR DESCRIPTION
**Pull request by victoralmau**
Change state fields to display translated by language

Please @pedrobaeza and @chienandalu can you review it?

@Tecnativa TT41343

### CODE REVIEW

File: stock_request/models/stock_request.py [hunk 1]
```diff
@@ -5,25 +5,20 @@
...
     _description = "Stock Request"
     _inherit = "stock.request.abstract"
     _order = "id desc"
 
-    def __get_request_states(self):
-        return REQUEST_STATES
-
-    def _get_request_states(self):
-        return self.__get_request_states()
+    def _get_states(self):
```
Comment (chienandalu) [inline stock_request/models/stock_request.py]: Why changing the method name?

File: stock_request/models/stock_request.py [hunk 2]
```diff
@@ -5,25 +5,20 @@
...
+        return [
+            ("draft", _("Draft")),
```
Comment (rousseldenis) [inline stock_request/models/stock_request.py]: You cannot include translations like that in selections. I have to check, but I've solved this in another way

File: stock_request/models/stock_request.py [hunk 3]
```diff
@@ -34,7 +29,7 @@ def _get_expected_date():
 
     name = fields.Char(states={"draft": [("readonly", False)]})
     state = fields.Selection(
-        selection=_get_request_states,
+        selection="_get_states",
```
Comment (rousseldenis) [inline stock_request/models/stock_request.py]: @victoralmau I think you should use a lambda method instead

Comment (rousseldenis) [reply stock_request/models/stock_request.py]: @victoralmau Please follow @chienandalu comments and change this to lambda? Thanks

**Comment by victoralmau** [comment]
Changes done according to another approach.

File: stock_request/models/stock_request_order.py [hunk 4]
```diff
@@ -42,7 +41,7 @@ def _get_default_requested_by(self):
         default="/",
     )
     state = fields.Selection(
-        selection=_get_request_order_states,
+        selection="_get_request_order_states",
```
Comment (rousseldenis) [inline stock_request/models/stock_request_order.py]: Could you change to lambda too ?

File: stock_request_submit/models/stock_request.py [hunk 5]
```diff
@@ -2,17 +2,17 @@
...
 
-    def __get_request_states(self):
-        states = super().__get_request_states()
-        if not ("submitted", "Submitted") in states:
+    def _get_request_states(self):
+        states = super()._get_request_states()
+        if not ("submitted", _("Submitted")) in states:
             states.insert(
-                states.index(("draft", "Draft")) + 1, ("submitted", "Submitted")
+                states.index(("draft", _("Draft"))) + 1, ("submitted", _("Submitted"))
```
Comment (chienandalu) [inline stock_request_submit/models/stock_request.py]: In any case, if this is the reason of this implementation, there's already a way to it without such overhead:

https://www.odoo.com/documentation/16.0/developer/reference/backend/orm.html?highlight=selection_add#odoo.fields.Selection
```
selection = [('a', 'A'), ('b', 'B')]
selection_add = [('c', 'C'), ('b',)]
> result = [('a', 'A'), ('c', 'C'), ('b', 'B')]
```

Another issue is that you need to mantain the uninstall hook, wich is also covered with `ondelete` parameter.

Comment (rousseldenis) [reply stock_request_submit/models/stock_request.py]: @chienandalu This is not available in v14

Comment (chienandalu) [reply stock_request_submit/models/stock_request.py]: Ok, then :sweat_smile:

Comment (pedrobaeza) [reply stock_request_submit/models/stock_request.py]: `selection_add` is in v14 AFAIK, and the uninstall thing is independent from using a selection method.

Comment (victoralmau) [reply stock_request_submit/models/stock_request.py]: Ok, changes done using this approach (`selection_add`) and simplifying everything.

**Review by chienandalu — APPROVED** [review]
:+1:

**Review by pedrobaeza — APPROVED** [review]
/ocabot merge minor

**Pull request merged**